### ISSUE TITLE
Fat line clipping misses intersection between curves

### ISSUE THREAD

**Issue by iconexperience** [paperjs/paper.js#1088]
Here is an interesting edge case where fat line clipping fails to find the intersection between two curves.

```javascript
var p1 = new paper.Path({segments:[ 
    [314.48576574722523, 308.77472829145444, 8.651222275575492, -2.220498935022104, -8.651222275575492, 2.220498935022105],
    [290.40724577164383, 302.0497467665447, 6.2486166726329975, 6.3819316872198515, -0.012648799703259704, -0.012918663419550348]
    ], closed:false})
var p2 = new paper.Path({segments:[
    [328.02746577749963, 300.1256389868014, 7.902769748335459, -10.046794724901815, -2.5352164811920375e-11, 3.2230218494078144e-11],
    [309.7443037372522, 309.63202128991924, 5.4001247917767614e-11, -2.9558577807620168e-12, -6.666656713343759, 0.3629571641783996], 
    ], closed:false})
var ints = p1.getIntersections(p2);
console.log(ints.length);
```

<a href="http://sketch.paperjs.org/#S/fZPBbtswDIZfxfApARxBpEiRTLHTTrsN6LHJwUvVpKhrB7axHYq+++i0GYZ1mwwYNvmL/PXRfqn79rnU2/r2qcyHU93Uh+F+ed/139uxOkP1qerLj+rcnssYvrbzafUyleNz6edpe1ft+srXXQIKpCyZhQSRMTVVihpkeVU0ICaiptKQGdCXMAuTYVNtMCBGMrXEERGiyzZ/0f0p433z3hwtBoqCxCKQKemlOQZXC2XJ2XtLU+WApBlyFswJzYSXWFKwBFkFwZTBY5sYImAmFdfEhOx3uoYNNOdEYMwxke7fLOyb6tANU7nfPrTdVF7X7/Dwv/Cu7FBDRHfq9oXM8sV+DICck5pmjeD9JZirsh9JU2Jic0vgnvyE5pDJIujFPgZOjA5CAQxjEi4bAK/pAB0LKJnTUiBa4r8opmhBiJJvSOITxMWFBUfle1DNvJjbYEftHEgMRBwu0Ft1b+tM/BMQjZIxOlJPLPPNIfviLJASuRs3HkPKaLxMC0STn9kJ/hPlGcI0j8NT+Tx0w+hId/VxLKXf1TeexA/Jsdy/pa77bg9t99gfPXmp+fuuj6nlWob36DPysBc5lvlLP5dxKof5cein1RnXLjz449CV0A3H1SIOXemP82l94//Qt7G0T+dhCdfbu/3rTw==">Here is the sketch</a>

The intersection is quite obvious and does not look like an edge case at all:

![image](https://cloud.githubusercontent.com/assets/2122773/16227735/c3866d9e-37b2-11e6-84e5-086ae920b775.png)

The only thing that I have found out so far is that if we set the tightening `factor` to `1` in `getCurveIntersections()`, the intersection is found.

**Comment by iconexperience** [comment]
One thing I noticed about our fatline clipping is that we always alternate between the two curves with every new call to `addCurveIntersections()`, even if the `t` values on one curve are already very close together. I stopped the alternation once the distance between `tMax`and `tMin` is less than `CLIPPING_EPSILON` on one curve and voilà, the intersection is found and the unit tests are all passed. I will test this a little more and if this does not introduce any new problems I will issue a PR.
